hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSClient.java

```diff
@@ -3159,6 +3165,21 @@ private void initThreadsNumForStripedReads(int numThreads) {
     }
   }
 
+  private void initBufferPoolForStripedReads(boolean useWeakReference) {
+    if (STRIPED_READ_BUFFER_POOL != null) {
+      return;
+    }
+    synchronized (DFSClient.class) {
```

Comment:
@Hexiaoqiao thanks for review. For this block, it's sort of modeled after other examples in DFSClient, such [as initializing the striped read thread pool](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSClient.java#L3150). I think the idea is that DFSClient could easily be used in multiple threads, so we want to avoid double initializing the shared resource.

Reply:
Got it.